R/map.R

```diff
@@ -211,12 +211,9 @@ map_ <- function(.type,
 
 mmap_ <- function(.x, .f, .progress, .type, error_call, ...) {
 
-  if (is.null(the$packages_installed)) {
-    rlang::check_installed(c("mirai", "carrier"), reason = "for parallel map.")
-    the$packages_installed <- TRUE
-  }
+  check_installed(c("mirai", "carrier"), reason = "for parallel map.")
```

Comment:
Do you want to check the versions here too? The version dependencies in suggests are not enforced by `install.packages()`.

Reply:
Good point - added version checks in [afccb43](https://github.com/tidyverse/purrr/pull/1181/commits/afccb439873a2d333f7e7cc86a074580b7d4c766). And because of that moved back to caching the result as we're no longer on the fast path for `rlang::check_installed()`.